### ISSUE TITLE
Duplicate game codes

### ISSUE THREAD

**Issue by styxit** [lmmfranco/nintendo-switch-eshop#31]
There are two games which return the same game code after parsing it with the `/HAC\w(\w{4})/` regex from [lib/constants.ts#L4](https://github.com/lmmfranco/nintendo-switch-eshop/blob/master/lib/constants.ts#L4).

The (US and EU) regex to parse the game codes like `HACXXXXX` return the same game code for two different games.

### Ludomania - released Aug 31, 2018
https://www.nintendo.com/games/detail/ludomania-switch
US game API returns code `HACNAQVMA`.
After applying the regex, this results in `AQVM`

### Flowlines VS - Released Jan 25, 2019
https://www.nintendo.com/games/detail/flowlines-vs-switch
US game API returns code `HACNAQVMB`.
After applying the regex, this results in `AQVM`

These now have the same parsed game code. The difference in the original code is the `A` and `B` at the end.

The Flowlines game has only been released a few weeks ago, but this looks to me that this might occur more often.

I'm working on something similar and wondered if you have encountered something like this before Looks like the regex might need an update.

**Comment by Favna** [comment]
I've brought this up with @lmmfranco to look into it

**Comment by lmmfranco** [comment]
Hi @styxit.
I've looked into that.
The original intention behind extracting the "parsed" game code was to find some common link between games on american, european and asian APIs.

I did some testing and this is what I found:

- Using 4 digits there are 53 games consistent between all regions ([json](https://pastebin.com/vBbwB3TG))
- Using 5 digits there are 38 games consistent between all regions ([json](https://pastebin.com/ZUnEVGW6))

If we take japan eshop out of the equation we have this:

- Using 4 digits there are 834 games consistent between US and EU regions
- Using 5 digits there are 797 games consistent between US and EU regions

Despite this, I still checked which codes are duplicate given the same region. That was the result:

Duplicated 4-digit code by region:
```js
{
    us: [ 'AAAB', 'AB38', 'AB38', 'AGBL', 'AQNY', 'BAAN' ],
    eu: [ 'AJ9K', 'ALJ2', 'AQVM' /* Flowlines */ ],
    jp: [ 'PADS', 'PADS' ]
}
```

Duplicated 5-digit code by region:
```js
{
    us: [ 'AB38A', 'AB38A', 'AGBLA', 'AQNYA', 'BAANA' ],
    eu: [ 'AJ9KA', 'ALJ2A' ],
    jp: []
}
```
Having said this, I think 4 digit game code is still better for cross-referencing games between regions, but should not be used as primary key for games.
